src/runway.rs

```diff
@@ -580,7 +592,7 @@ where
 
     async fn on_create(&mut self, u: PreUnit<H>) {
         debug!(target: "AlephBFT-runway", "{:?} On create notification.", self.index());
-        let data = self.data_io.get_data();
+        let data = self.data_provider.get_data();
```

Comment:
Open question, no need to answer: why it's not an `async` method? Looks a bit scary considering a single threaded executor and some magic that might be happening behind the scene.

Reply:
Very good point, it probably should be and this is a good opportunity to change it.

Reply:
Added to both `get_data` and `data_finalized`